plugins/inventory/aws_ec2.py

```diff
@@ -505,8 +508,16 @@ def _get_hostname(self, instance, hostnames):
 
         hostname = None
         for preference in hostnames:
-            if 'tag' in preference:
-                if not preference.startswith('tag:'):
+            if isinstance(preference, dict):
+                if 'name' not in preference:
+                    raise AnsibleError("A 'name' key must be defined in a hostnames dictionary.")
+                hostname = self._get_hostname(instance, [preference["name"]])
+                hostname_from_prefix = self._get_hostname(instance, [preference["prefix"]])
+                separator = preference.get("separator", "_")
+                if hostname and hostname_from_prefix and 'prefix' in preference:
+                    hostname = hostname_from_prefix + separator + hostname
+            elif preference.startswith('tag:'):
+                if not re.match(r'^tag:\S+=\S+$', preference):
```

Comment:
The tests are failing because of this test since `tag:Name` is not accepted anymore. I think we can just just drop these two lines.

Reply:
Dropping this this `elif` condition block? What do you mean by `tag:Name` no longer supported?

Reply:
Just the line 520 and 521. The regex does not match `tag:Name`.

Reply:
For clarity, this is what I mean:

```patch
diff --git a/plugins/inventory/aws_ec2.py b/plugins/inventory/aws_ec2.py
index c382f34..f217e79 100644
--- a/plugins/inventory/aws_ec2.py
+++ b/plugins/inventory/aws_ec2.py
@@ -517,8 +517,6 @@ class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable):
                 if hostname and hostname_from_prefix and 'prefix' in preference:
                     hostname = hostname_from_prefix + separator + hostname
             elif preference.startswith('tag:'):
-                if not re.match(r'^tag:\S+=\S+$', preference):
-                    raise AnsibleError("To name a host by tags name_value, use 'tag:name=value'.")
                 hostname = self._get_tag_hostname(preference, instance)
             else:
                 hostname = self._get_boto_attr_chain(preference, instance)
```

Reply:
Thanks. Note also that there is a feature in GitHub to select multiple lines and then add a comment for them.

Reply:
:-) Thanks for the suggestion.